src/alchemlyb/parsing/amber.py

```diff
@@ -65,8 +65,9 @@ def __init__(self, filename):
         self.filename = filename
         try:
             self.fileh = anyopen(self.filename, 'r')
-        except Exception as ex:  # pragma: no cover
+        except Exception as ex:
             logger.exception("Cannot open file %s", filename)
+            raise ex
```

Comment:
IIRC, this can just be 
```suggestion
            raise
```
and it will re-raise the exception itself.

We can then simplify the line above, too.